components/scale/dateTime.js

```diff
@@ -1,401 +1,65 @@
-    * the real value. This generally produces crisper graphics.
-    *
-    * @type {object}
-    * @memberof fc.scale
-    * @class fc.scale.dateTime
-    */
     fc.scale.dateTime = function() {
         return dateTimeScale();
     };
 
-    /**
-    * Constructs a new instance of the dateTime scale component.
-    *
-    * @memberof fc.scale.dateTime
-    * @param {d3.scale.linear} linear used in the copy constructor to copy the base linear
-    * scale between the original and the copy.
-    * @param {array} baseDomain used in the copy constructor to copy the base domain (Max
-    * and Min) between the original and the copy.
-    * @param {boolean} alignPixels used in the copy constructor to copy the pixel alignment
-    * option between the original and the copy.
-    * @param {boolean} hideWeekends used in the copy constructor to copy the hide weekends
-    * option between the original and the copy.
-    */
-    function dateTimeScale(linear, baseDomain, alignPixels, hideWeekends, padEnds) {
+
+    function dateTimeScale(adaptedScale, discontinuityProvider) {
 
         if (!arguments.length) {
-            linear = d3.scale.linear();
-            baseDomain = [new Date(0), new Date(0)];
-            alignPixels = true;
-            hideWeekends = false;
-            padEnds = false;
+            adaptedScale = d3.time.scale();
+            discontinuityProvider = fc.scale.discontinuity.identity;
         }
 
-        /**
-        * Used to scale a value from domain space to pixel space. This function is used primarily
-        * to position elements on the X axis.
-        *
-        * @memberof fc.scale.dateTime
-        * @method scale
-        * @param {object} x the real world domain value to be scaled.
-        * @returns the converted value in pixel space. This value is also pixel aligned if the
-        * relevant options are set.
-        */
-        function scale(x) {
-            var n = 0;
-            if (typeof x === 'number') {
-                // When scaling ticks.
-                n = linear(x);
-            } else {
-                // When scaling dates.
-                n = linear(linearTime(x));
-            }
-            return alignPixels ? Math.round(n) : n;
+        function discontinuities() { return scale.discontinuityProvider.value; }
+
+        function scale(date) {
+            var domain = adaptedScale.domain();
+            var range = adaptedScale.range();
+
+            // The discontinuityProvider is responsible for determine the distance between two points
+            // along a scale that has discontinuities (i.e. sections that have been removed).
+            // the scale for the given point 'x' is calculated as the ratio of the discontinuous distance
+            // over the domain of this axis, versus the discontinuous distance to 'x'
+            var totalDomainDistance = discontinuities().getDistance(domain);
```

Comment:
With either the identity or noop default implementations it looks like this line will error?